src/ray/common/status_or.h

```diff
@@ -0,0 +1,257 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <stdexcept>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+
+#include "absl/base/attributes.h"
+#include "ray/common/status.h"
+
+namespace ray {
+
+// TODO(hjiang): Introduce statusor related macros.
+template <typename T>
+class StatusOr {
+ public:
+  StatusOr() noexcept = default;
+  // NOLINTNEXTLINE(runtime/explicit)
+  StatusOr(Status status) : status_(std::move(status)) {}
+  // NOLINTNEXTLINE(runtime/explicit)
+  StatusOr(const T &data) : status_(Status::OK()), data_(data) {}
+  // NOLINTNEXTLINE(runtime/explicit)
```

Comment:
curious what this is for?

Reply:
Used to suppress clang-tidy complaint.

Reply:
Checkout https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics